spec/worker_spec.rb

```diff
@@ -0,0 +1,139 @@
+RSpec.describe ActiveAdmin::AsyncExporter::Worker do
+  let!(:users) do
+    %w[user_1 user_2 user_3].map do |username|
+      User.find_or_create_by!(username: username)
+    end
+  end
+  let(:controller) { Admin::UsersController.new }
+  let(:decorate_model) { false }
+  let(:file_name) { 'file_name' }
+  let(:query) { nil }
+  let(:csv_path) { './csv' }
+  let(:admin) { User.find_or_create_by!(username: 'AdminUser') }
+  let(:csv_fixture_file_path) { "./spec/fixtures/files/#{csv_fixture_file_name}.csv" }
+  let(:csv_fixture_file_name) { 'username' }
+
+  let(:admin_report) do
+    AdminReport.create!(author_id: admin.id,
+                        entity: User.name,
+                        status: :pending)
+  end
+
+  let(:columns) { { username: { block: false, value: 'username' } } }
+
+  let(:options) do
+    {
+      admin_report_id: admin_report.id,
+      controller: controller.class.name,
+      columns: columns,
+      decorate_model: decorate_model,
+      file_name: file_name,
+      query: query
+    }
+  end
+
+  subject do
+    described_class.perform_now(options)
+  end
+
+  context 'with valid options' do
+    let(:test_date) { Time.current }
+    let(:full_file_name) { [file_name, test_date.to_i].join('_') + '.csv' }
+    let(:file_path) { "#{csv_path}/#{full_file_name}" }
+
+    before do
+      Timecop.freeze(test_date)
+    end
+
+    after do
+      FileUtils.rm(file_path)
+      Timecop.return
+    end
+
+    shared_examples 'creates the physical file with data' do
+      it 'creates the physical file at the given location_url' do
+        subject
+        expect(File.open(admin_report.reload.location_url)).to be
```

Comment:
Would this be more descriptive? 
```suggestion
        expect { subject }.to change { admin_report.reload.location_url }
```

Reply:
Definitely! But this would be testing something else, right? On the given test I am testing that the file exist, whereas your suggestion tests the AdminReport location_url. I believe this suggestion would apply more to https://github.com/rootstrap/activeadmin-async_exporter/pull/22/files#diff-08cb32684eff0fd6876c676b821fc3a3519e8a3addbb5281c0e320ef60ff6e80R82-R84